client/components/timezone/index.jsx

```diff
@@ -15,9 +16,32 @@ import QueryTimezones from 'components/data/query-timezones';
 import getRawOffsets from 'state/selectors/get-raw-offsets';
 import getTimezones from 'state/selectors/get-timezones';
 
+function maybeFindLinkedTimezone( timezone, timezones ) {
+	if ( ! timezones || timezone in timezones ) {
+		return timezone;
+	}
+	// The (guessed) timezone could not be found, so let's see if a linked timezone for this exists.
+	// Example: we don't have Asia/Calcutta but Asia/Kalkota, so we'll need to choose that.
+	timezone = reduce(
+		pickBy( flattenDepth( timezones, 2 ), isArray ),
+		function( mappedTimezone, option ) {
+			const [ value ] = option;
+			// This uses the internal linking array of moment-timezone since the links are not exposed otherwise.
+			const mapping = moment.tz._links[ value.toLowerCase().replace( /\//g, '_' ) ];
+			if ( mapping && moment.tz.zone( mapping ).name === mappedTimezone ) {
+				return value;
+			}
+			return mappedTimezone;
+		},
+		timezone
+	);
+
+	return timezone;
+}
+
 class Timezone extends Component {
 	onSelect = event => {
-		this.props.onSelect( event.target.value );
+		this.props.onSelect( maybeFindLinkedTimezone( event.target.value, this.props.timezones ) );
```

Comment:
This normalization shouldn't be needed. `event.target.value` is the DOM value that was rendered in `<select value={ ... }>`, and that's always already normalized.